### DISCUSSION TITLE
Splitter resize cause panel rerender

### DISCUSSION THREAD

**Discussion by ianclough** [microsoft/fluentui-blazor#1772] [General]
Resizing the Splitter panes using the splitter bar causes the panel and its child components to be re-rendered, this doesn't happen if the browser window itself is re-sized. Is there any intention to prevent this or provide such a configuration option? I can probably work around this, one of my panel is quite expensive to re-render.

**Comment by vnbaaij** [comment]
I don't think there is a way we can prevent the re-render on moving the bar. If anyone knows of a way, I'm keen to hear about it.

**Comment by ianclough** [comment]
There has been quite a bit of discussion on this topic https://github.com/dotnet/aspnetcore/issues/18919 but handling this in the application seems to the only viable solution currently.

**Reply by vnbaaij** [reply]
I think so too. We don't know /have no control over what gets displayed in the panels so we can't utilize ShouldRender on our end.

**Reply by ianclough** [reply]
I have done some more investigation. Simply clicking on the splitter bar without moving it causes 3 or 4 renders of the panel. Removing the event handlers, onsplittercollapsed, onsplitterresized, prevents the panel(s) from re-rendering. Would it be a useful enhancement to conditionally attach these event handles if the respective parameter is present? I was thinking either conditional razor markup or custom rendering.

**Reply by vnbaaij** [reply]
You can't conditionally render `@on...` handlers

**Reply by ianclough** [reply]
Yes, I was thinking of either making the whole split-panels element conditional one with and one without event handlers or probably better overriding BuildRenderTree.
